src/views/FormView.vue

```diff
@@ -0,0 +1,182 @@
+        </Form>
+        <div>
+            <RouterLink to="/movies">
+                <button>Go to Movies App</button>
+            </RouterLink>
+        </div>
+    </div>
+</template>
+
+<script setup lang="ts">
+import { ref, computed } from 'vue';
+import { Form, Field, ErrorMessage } from 'vee-validate';
+import { object, string, date } from 'yup';
+
+const genders = ['Mr', 'Mrs', 'Miss', 'Ms']
+const employmentStatuses = ['Full time', 'Part time', 'Unemployed', 'Retired']
+
+const numberOnlyRegex = /^\d+$/
+const mobilePhoneRegex =
+    /^(?:(?:\+\s*(?:4\s*){2}[1|2]|0\s*[1|2])\s*(?:\d\s*){8,9}|(?:\+\s*(?:4\s*){2}7|0\s*7)\s*(?:\d\s*){9})$/ // GB number like: 07111111111
+
+const userNameRegex = /^(?:[a-zA-Z][a-zA-Z\s\\.\-']{1,29})$/ // at least 2 characters, and can only include letters, apostrophes, hyphens and full stops
+
+const numberOfDependants = ref<number>();
+const childCareCostsVisible = computed(() => showChildCareCosts(numberOfDependants.value));
+
+const employmentStatus = ref<string>('');
+const companyNameVisible = computed(() => showCompanyName(employmentStatus.value));
+
+const showChildCareCosts = (numberOfDependants: number | undefined = 0) => numberOfDependants > 0;
+const showCompanyName = (companyName: string) => companyName === 'Full time' || companyName === 'Part time';
+
+const maxDate = new Date();
+maxDate.setFullYear(maxDate.getFullYear() - 18);
+
+const minDate = new Date();
+minDate.setFullYear(minDate.getFullYear() - 100);
+
+const schema = object({
+    firstName: string().required('First name is required').matches(userNameRegex, 'Invalid format'),
+    lastName: string().required('Last name is required').matches(userNameRegex, 'Invalid format'),
+    gender: string().required('Gender is required').oneOf(genders),
+    birthDate: date().required('Birth date is required').min(minDate, 'Must be under 100 years old').max(maxDate, 'Must be over 18 years old'),
+    mobilePhone: string().required('Mobile Phone is required').matches(mobilePhoneRegex, 'Invalid format. GB number like: 07111111111.'),
+    numberOfDependants: string().matches(numberOnlyRegex, 'Invalid format. Should be number and greater than 0.'),
+    childCareCosts: string().when('numberOfDependants', {
+        is: showChildCareCosts,
+        then: (schema) => schema.required('Child care cost is required').matches(numberOnlyRegex, 'Invalid format. Should be number.').max(1000000, 'Must be below 1,000,000'),
+        otherwise: (schema) => schema
+    }),
+    employmentStatus: string().required('Employemnt status is required').oneOf(employmentStatuses),
+    companyName: string().when('employmentStatus', {
+        is: showCompanyName,
+        then: (schema) => schema.required('Company Name is required'),
+        otherwise: (schema) => schema
+    }),
+    personalIncome: string().matches(numberOnlyRegex, 'Invalid format. Should be number.').min(1000, 'Must be above 1000').max(1000000, 'Must be below 1,000,000'),
```

Comment:
- 1 - Must be above 1000
- 999 - Must be above 1000
- 10000 - Must be above 1000
- 1000000000 - Must be above 1000

Reply:
fixed